File: 49-group-anagrams/group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        vector<vector<string>> ans;
        map<string, vector<string>>mp;
        vector<string> temp = strs;
        for(int i = 0; i<temp.size(); i++){
            string val = temp[i];
            sort(temp[i].begin(), temp[i].end());
            mp[temp[i]].push_back(val);
        }
        for(auto i : mp){
            ans.push_back(i.second);
        }
        return ans;
    }
};

File: 49-group-anagrams/group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        ans = []
        mp = {}
        temp = strs[:]
        
        for i in range(len(temp)):
            val = temp[i]
            temp[i] = ''.join(sorted(temp[i]))
            if temp[i] in mp:
                mp[temp[i]].append(val)
            else:
                mp[temp[i]] = [val]
        
        for key, value in mp.items():
            ans.append(value)
        
        return ans
